src/core/qgstemporalrangeobject.h

```diff
@@ -0,0 +1,102 @@
+/***************************************************************************
+                         qgstemporalrangeobject.h
+                         ---------------
+    begin                : January 2020
+    copyright            : (C) 2020 by Samweli Mwakisambwe
+    email                : samweli at kartoza dot com
+ ***************************************************************************/
+
+/***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+
+#ifndef QGSTEMPORALRANGEOBJECT_H
+#define QGSTEMPORALRANGEOBJECT_H
+
+#include "qgis_core.h"
+#include "qgis_sip.h"
+#include "qgsrange.h"
+
+class QgsRenderContext;
+
+/**
+ * \class QgsTemporalRangeObject
+ * \ingroup core
+ * Base class for temporal based classes.
+ *
+ * Subclasses may wish to update the abstract temporal functions.
+ *
+ * \since QGIS 3.14
+ */
+class CORE_EXPORT QgsTemporalRangeObject
+{
+  public:
+
+    /**
+     * Constructor for QgsTemporalRangeObject.
+     */
+    QgsTemporalRangeObject();
+
+    /**
+     * Creates temporal range object with the specified temporal state.
+     */
+    QgsTemporalRangeObject( bool enabled );
+
+    /**
+     * Copies render context temporal members
+     */
+    QgsTemporalRangeObject( const QgsRenderContext &rh );
```

Comment:
Not required

Reply:
If we remove this we wont be able to copy over the render context temporal attributes here
https://github.com/qgis/QGIS/blob/96c8c2f7d2046c6db5b9204b97ea37ab7d012ba1/src/core/qgsrendercontext.cpp#L42

Reply:
Can you backport this change from your other PR, and then we'll merge this one? It'll make #34590 smaller and easier to review as a result...

Reply:
Alright